openedx/core/djangoapps/content_libraries/views_collections.py

```diff
@@ -0,0 +1,184 @@
+"""
+Collections API Views
+"""
+
+from __future__ import annotations
+
+from django.db.models import QuerySet
+
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.status import HTTP_405_METHOD_NOT_ALLOWED
+
+from opaque_keys.edx.locator import LibraryLocatorV2
+from openedx_learning.api import authoring as authoring_api
+from openedx_learning.api.authoring_models import Collection
+
+from openedx.core.djangoapps.content_libraries import api, permissions
+from openedx.core.djangoapps.content_libraries.models import ContentLibrary
+from openedx.core.djangoapps.content_libraries.views import convert_exceptions
+from openedx.core.djangoapps.content_libraries.serializers import (
+    ContentLibraryCollectionSerializer,
+    ContentLibraryCollectionComponentsUpdateSerializer,
+    ContentLibraryCollectionCreateSerializer,
+    ContentLibraryCollectionUpdateSerializer,
+)
+
+
+class LibraryCollectionsView(ModelViewSet):
+    """
+    Views to get, create and update Library Collections.
+    """
+
+    serializer_class = ContentLibraryCollectionSerializer
+    lookup_field = 'key'
```

Comment:
@pomegranited Since we plan to [make use](https://github.com/openedx/frontend-app-course-authoring/issues/1101#issuecomment-2300957646) of `object_key` i.e. `LibraryCollectionKey` in frontend urls, we should also convert this view to accept `LibraryCollectionKey` instead of `key` similar to [libraries](https://github.com/open-craft/edx-platform/blob/d312ff3e6aa1f02785f2cd2dd68d54df02359ad1/openedx/core/djangoapps/content_libraries/views.py#L290-L299). **Provided** we really want to include `LibraryCollectionKey` in frontend url.

Reply:
Same, there are different places where `LibraryCollectionKey` can be used. I agree with creating a task for it and doing it when opaque-keys is [updated in edx-platform](https://tasks.opencraft.com/browse/FAL-3785?focusedCommentId=296651&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-296651)

Reply:
For now I would prefer to not use the opaque keys in too many places, other than for tagging. We actually tried to avoid using the opaque key completely, except that it's required for tagging. As I mentioned in https://github.com/openedx/edx-platform/pull/35321#discussion_r1742724186 for the URLs and APIs we can generally just accept both a library ID and a `key`/slug (not the full opaque key, just the key part) as separate URL parameters. The python APIs can just use the collection integer IDs which is more efficient than huge string keys anyways.

Reply:
@bradenmacdonald Yes, that is much better.